scripts/add-icon.js

```diff
@@ -0,0 +1,71 @@
+import fs from 'node:fs/promises';
+import inquirer from 'inquirer';
+import chalkPipe from 'chalk-pipe';
+import logSymbols from 'log-symbols';
+import getRelativeLuminance from 'get-relative-luminance';
+import { getIconsDataString, getIconDataPath, titleToSlug } from './utils.js';
+
+const hexPattern = /^#?[a-fA-F0-9]{6}$/;
+const iconsData = JSON.parse(await getIconsDataString());
+
+const prompts = [
+  {
+    type: 'input',
+    name: 'title',
+    validate: (text) => {
+      if (!text) return 'This field is required';
+      if (iconsData.icons.find((x) => x.title === text))
+        return 'This icon title already exist';
+      return true;
+    },
+    message: 'Title',
+  },
+  {
+    type: 'input',
+    name: 'hex',
+    message: 'Hex',
+    validate: (text) =>
+      hexPattern.test(text) ? true : 'This should be a 6-digit hex code',
+    transformer: (text) => {
+      const color = text.startsWith('#') ? text : `#${text}`;
+      const luminance = hexPattern.test(text)
+        ? getRelativeLuminance.default(color)
+        : -1;
+      if (luminance === -1) return text;
+      return chalkPipe(`bg${color}.${luminance < 0.4 ? 'white' : 'black'}`)(
+        text,
+      );
+    },
+  },
+  {
+    type: 'input',
+    name: 'source',
+    message: 'Source',
+    validate: (text) =>
+      Boolean(text.startsWith('https://') || text.startsWith('http://'))
+        ? true
+        : 'This should be a URL',
+  },
+];
+
+const icon = await inquirer.prompt(prompts);
+
+iconsData.icons.push({
+  title: icon.title,
+  hex: icon.hex.replace(/^#/, '').toUpperCase(),
+  source: icon.source,
+});
+iconsData.icons.sort((a, b) => a.title.localeCompare(b.title));
+
+await fs.writeFile(
+  getIconDataPath(),
+  JSON.stringify(iconsData, null, 4) + '\n',
```

Comment:
This is not consistent with our JSON prettification linter. Please, mimic the implementation already defined in *scripts/lint/ourlint.js*: https://github.com/simple-icons/simple-icons/blob/d5444f699528a456ee2894751994b4eef7274235/scripts/lint/ourlint.js#L50-L51

Newlines must be normalized because Windows users could read the data file using `\r\n` newlines.

Reply:
`JSON.stringify(data, null, 4)` and `JSON.stringify(data, null, '    ')` are the same. And I have a `\n` at the end of the file.

`JSON.stringify`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

But I change my code to `${JSON.stringify(data, null, '    ')}\n` for making our code consistent.

Reply:
Maybe would be worth to add a new utility function `writeIconsData` to normalize all calls.